src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbQueryIndexSelector.java

```diff
@@ -24,19 +24,19 @@
      * @param availableIndexes list of possible DynamoDB indexes
      * @return most restricted index or null if non did match the criteria
      */
-    public DynamodbIndex findMostRestrictedIndex(final QueryPredicate<DynamodbNodeVisitor> selection,
-            final List<DynamodbIndex> availableIndexes) {
-        final List<DynamodbIndex> suitableKeys = extractSuitableIndexes(selection, availableIndexes);
+    public AbstractDynamodbIndex findMostRestrictedIndex(final QueryPredicate<DynamodbNodeVisitor> selection,
+            final List<AbstractDynamodbIndex> availableIndexes) {
```

Comment:
I don't really like that you have a list of abstract class's descendants. Maybe you can also create an interface for these classes?